src/Lean/Server/Completion.lean

```diff
@@ -274,7 +274,10 @@ private def tacticCompletion (ctx : ContextInfo) : IO (Option CompletionList) :=
       items.push { label := tk.toString, detail? := none, documentation? := none }
     return some { items := sortCompletionItems items, isIncomplete := true }
 
-partial def find? (fileMap : FileMap) (hoverPos : String.Pos) (infoTree : InfoTree) : IO (Option CompletionList) := do
+partial def find? (fileMap : FileMap) (hoverPos : String.Pos) (infoTree : InfoTree) (trigger : CompletionTriggerKind): IO (Option CompletionList) := do
+  -- TODO: When user types CTRL+SPACE in "Append.app|end" with the cursor placed in the middle of the word like
+  -- that then the trigger is CompletionTriggerKind.Invoke and in that case it is supposed to return all
+  -- CompletionItems matching to the left of the cursor, so "app" in this example.
```

Comment:
We should file an issue for this

Reply:
Looks like you did, thanks: https://github.com/leanprover/lean4/issues/857